### ISSUE TITLE
Build Arguments stop to pass into Dockerfile

### ISSUE THREAD

**Issue by Antonytm** [mr-smithers-excellent/docker-build-push#102]
My Github action:

```
name: "Publish to Docker"
description: "Pushes built artifacts to Docker"
inputs:
  build_configuration:
    description: "Debug/Release"
    required: true
  dotnet_version:
    description: "net45/net46/net471/net48/etc"
    required: true
  username:
    description: "Docker registry username"
    required: true
  password:
    description: "Docker registry password"
    required: true
runs:
  using: "composite"
  steps:
  - uses: mr-smithers-excellent/docker-build-push@v5.6
    name: Build & push Docker image
    with:
      image: antonytm/dianoga-assets
      tags: 6.0.0-beta.2-${{ inputs.dotnet_version }}-${{ inputs.build_configuration }}-${{ github.run_number }}, latest-${{ inputs.dotnet_version }}-${{ inputs.build_configuration }}
      registry: docker.io
      dockerfile: Dockerfile
      username: ${{ inputs.username }}
      password: ${{ inputs.password }}
      buildArgs: BUILD_CONFIGURATION=${{ inputs.build_configuration }}, DOTNET_VERSION_ARG=${{ inputs.dotnet_version }}
```

It does work 4 months ago. But now it stopped.
Was anything changed in the way how arguments should be passed?

Link to the build:
https://github.com/Antonytm/Dianoga/runs/7138369191?check_suite_focus=true

**Comment by mr-smithers-excellent** [comment]
Hi @Antonytm - nothing has changed with the 5.6 release. It looks like this error is actually an issue with a build step within your Dockerfile. Have you tried this same docker build locally?

**Comment by Antonytm** [comment]
I made a fork from this repository:
And run 2 tests with passing build arguments:

**Windows:**

https://github.com/Antonytm/docker-build-push/runs/7139620913?check_suite_focus=true

Windows locally:
```
PS C:\source\docker-build-push\e2e> docker build . --build-arg TEST1=test --no-cache
Sending build context to Docker daemon  4.608kB
Step 1/6 : FROM mcr.microsoft.com/dotnet/framework/sdk:4.8
 ---> f548dad92f88
Step 2/6 : ARG TEST1
 ---> Running in 2f96de372a2f
Removing intermediate container 2f96de372a2f
 ---> eba6856a7fa9
Step 3/6 : ARG TEST2
 ---> Running in a01867426933
Removing intermediate container a01867426933
 ---> 874a6e5d854a
Step 4/6 : ARG TEST3
 ---> Running in 2a2a34195e72
Removing intermediate container 2a2a34195e72
 ---> 4d67e7d608a5
Step 5/6 : RUN echo ${TEST1}
 ---> Running in ae35f0606efc
Removing intermediate container ae35f0606efc
 ---> a9aa485b0fd9
Step 6/6 : RUN echo $env:TEST1
 ---> Running in 85d72fef9c14
test
Removing intermediate container 85d72fef9c14
 ---> eb324f72d88a
Successfully built eb324f72d88a
```

**Linux:**
https://github.com/Antonytm/docker-build-push/runs/7139449917?check_suite_focus=true

Linux locally:
```
PS C:\source\docker-build-push\e2e> docker build . --platform linux/amd64 --build-arg TEST1=test --no-cache
[+] Building 1.7s (7/7) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                                         0.0s
 => => transferring dockerfile: 130B                                                                                                                                                                         0.0s
 => [internal] load .dockerignore                                                                                                                                                                            0.0s
 => => transferring context: 2B                                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/library/alpine:latest                                                                                                                                             1.5s
 => [auth] library/alpine:pull token for registry-1.docker.io                                                                                                                                                0.0s
 => CACHED [1/2] FROM docker.io/library/alpine@sha256:686d8c9dfa6f3ccfc8230bc3178d23f84eeaf7e457f36f271ab1acc53015037c                                                                                       0.0s
 => [2/2] RUN echo test                                                                                                                                                                                      0.2s
 => exporting to image                                                                                                                                                                                       0.0s
 => => exporting layers                                                                                                                                                                                      0.0s
 => => writing image sha256:5a6e8cc0f469dd147dc7dc6525a1b4c868cd15908b73ba15228e5f9cf9f77443
```

**Actual result:**
Echo of build argument works locally for both platforms: Windows and Linux
Echo of build argument in GitHub workflows works only for Linux

**Expected result:**
Echo of build argument works locally for both platforms: Windows and Linux
Echo of build argument in GitHub workflows works for both platforms: Windows and Linux

**Comment by Antonytm** [comment]
Hi @mr-smithers-excellent 

Sorry for the false alarm, there is no issue with [docker-build-push](https://github.com/mr-smithers-excellent/docker-build-push).

I wrote Github Docker build action without usage `docker-build-push` and got the same behavior.

I started to look deeper and found the root of the issue. Github actions behave differently depending on where you run them. (`jobs>build>run-on`). 

I run my build on `windows-latest` and nothing changed in a code for me, but there were changes in Github.
4 months ago: `windows-latest` was `window-2019`
Now: `windows-latest` is `windows-2022`

JFYI: If you face any issues building Windows images, the first thing to try should be switch to `windows-2019` and check.

Link to the reported issue:
https://github.com/actions/virtual-environments/issues/5857
